src/codeu/chat/common/DerbyDatabaseInteractions.java

```diff
@@ -0,0 +1,39 @@
+package codeu.chat.common;
+
+import codeu.chat.util.store.Store;
+
+import java.io.IOException;
+import java.sql.SQLException;
+
+import codeu.chat.util.Uuid;
+
+public interface DerbyDatabaseInteractions {
+	
+	// USER
+	public Store<Uuid, User> getAllUsers();
+	
+	public User getUser(Uuid userid) throws SQLException, IOException;
+	
+	public void addUser(User u) throws SQLException;
+	
+	public User userLogin(String email, String password)  throws SQLException, IOException;
```

Comment:
What exactly should this function do? Return the User if and only if the email and password match something in the database? What does it return if the login is wrong?